### PR TITLE
Fix issue with mothballer compiling

### PR DESCRIPTION
**Pull request by int00h5525**
See #38

### CODE REVIEW

File: tools/mothballer.py [hunk 1]
```diff
@@ -24,17 +24,19 @@ def write_kv_pairs(ziphandle, filename, kv):
     """
     filehandle = io.StringIO()
     for key in sorted(kv.keys()):
-        if type(kv[key])  == type([]):
+        if isinstance(kv[key], type([])):
```
Comment (nealey) [inline tools/mothballer.py]: let's do this right if we're going to fix it:
`if isinstance(kv[key], list):`

File: tools/mothballer.py [hunk 2]
```diff
@@ -101,14 +102,14 @@ def build_category(categorydir, outdir):
         # open and gather some state
         existing = zipfile.ZipFile(zipfilename, 'r')
         try:
-            category_seed = existing.open(seedfn).read().strip()
-        except:
+            category_seed = existing.open(SEEDFN).read().strip()
```
Comment (nealey) [inline tools/mothballer.py]: How could this have worked before? Is there a `seedfn` anywhere else in this file?

File: tools/mothballer.py [hunk 3]
```diff
@@ -101,14 +102,14 @@ def build_category(categorydir, outdir):
...
+        except Exception:
             pass
         existing.close()
     logging.debug("Using PRNG seed {}".format(category_seed))
 
     zipfileraw = tempfile.NamedTemporaryFile(delete=False)
-    mothball = package(categoryname, categorydir, zfraw)
+    mothball = package(categoryname, categorydir, category_seed)
```
Comment (nealey) [inline tools/mothballer.py]: This looks like you're switching from a file handle to a random number seed. What?

File: tools/mothballer.py [hunk 4]
```diff
@@ -161,8 +162,8 @@ def package(categoryname, categorydir, seed):
     zf.close()
     zfraw.seek(0)
     return zfraw
```
Comment (nealey) [inline tools/mothballer.py]: shouldn't this be `zipfileraw`?

File: tools/mothballer.py (hunk 2 above)
Comment (int00h5525) [reply tools/mothballer.py]: Nope. Nowhere.

File: tools/mothballer.py (hunk 3 above)
Comment (int00h5525) [reply tools/mothballer.py]: The `package` function's third argument is expected to be a seed value (as bytes), not a file object.

File: tools/mothballer.py (hunk 4 above)
Comment (int00h5525) [reply tools/mothballer.py]: `build_category` uses zipfileraw as a `NamedTemporaryFile`, to which it write the result of the `package` function, which uses `zfraw` to represent a file-like `BytesIO` object that has zip-formatted contents, which are generated by `zf`

File: tools/mothballer.py (hunk 3 above)
Comment (nealey) [reply tools/mothballer.py]: That is nuts.

File: tools/mothballer.py (hunk 4 above)
Comment (nealey) [reply tools/mothballer.py]: Okay that is a bad choice of variable names but you don't have to fix it in your merge.

**Pull request merged**